### PR TITLE
Fix minor warnings reported by the compiler or cppcheck

### PR DESCRIPTION
**Pull request by chbruyand**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/histog.hh [hunk 1]
```diff
@@ -21,7 +21,7 @@ struct LogHistogramBin
 
 template<typename T>
 std::vector<LogHistogramBin> createLogHistogram(const T& bins,
-                        std::deque<double> percentiles={0.001, 0.01, 0.1, 0.2, 0.5, 1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 94, 95, 96, 97, 97.5, 98, 98.5, 99, 99.5, 99.6, 99.9, 99.99, 99.999, 99.9999})
+                        std::deque<double>& percentiles={0.001, 0.01, 0.1, 0.2, 0.5, 1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 94, 95, 96, 97, 97.5, 98, 98.5, 99, 99.5, 99.6, 99.9, 99.99, 99.999, 99.9999})
```
Comment (rgacogne) [inline pdns/histog.hh]: I'm afraid binding a default value to a reference can't work, unless I'm mistaken. Note that this file was never actually compiled before https://github.com/PowerDNS/pdns/pull/7291.

File: pdns/histog.hh [hunk 2]
```diff
@@ -60,7 +60,7 @@ std::vector<LogHistogramBin> createLogHistogram(const T& bins,
 }
 
 template<typename T>
-void writeLogHistogramFile(const T& bins, std::ostream& out, std::deque<double> percentiles={0.001, 0.01, 0.1, 0.2, 0.5, 1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 94, 95, 96, 97, 97.5, 98, 98.5, 99, 99.5, 99.6, 99.9, 99.99, 99.999, 99.9999} )
+void writeLogHistogramFile(const T& bins, std::ostream& out, std::deque<double>& percentiles={0.001, 0.01, 0.1, 0.2, 0.5, 1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90, 94, 95, 96, 97, 97.5, 98, 98.5, 99, 99.5, 99.6, 99.9, 99.99, 99.999, 99.9999} )
 {
```
Comment (rgacogne) [inline pdns/histog.hh]: Same as above.

File: pdns/lazy_allocator.hh [hunk 3]
```diff
@@ -35,12 +35,12 @@ struct lazy_allocator {
                    "lazy_allocator must only be used with trivial types");
 
     pointer
-    allocate (size_type const n) {
+    allocate (size_type const& n) {
```
Comment (rgacogne) [inline pdns/lazy_allocator.hh]: I'm not sure passing a `size_t` by const reference is a good idea, is `cppcheck` really issuing a warning about that?

File: pdns/rcpgenerator.cc [hunk 4]
```diff
@@ -545,12 +545,7 @@ void RecordTextWriter::xfrTime(const uint32_t& val)
   time_t time=val; // Y2038 bug!
   gmtime_r(&time, &tm);
 
-  char tmp[16];
-  snprintf(tmp,sizeof(tmp)-1, "%04d%02d%02d" "%02d%02d%02d", 
-           tm.tm_year+1900, tm.tm_mon+1, tm.tm_mday, 
-           tm.tm_hour, tm.tm_min, tm.tm_sec);
-  
-  d_string += tmp;
+  d_string += boost::str(boost::format("%04d%02d%02d" "%02d%02d%02d") % (tm.tm_year+1900) % (tm.tm_mon+1) % tm.tm_mday % tm.tm_hour % tm.tm_min % tm.tm_sec);
```
Comment (rgacogne) [inline pdns/rcpgenerator.cc]: The only minor worry I have with this change is the performance impact. Perhaps we could do something like this, as boost's documentation suggests it should be slightly faster:
```suggestion
  static const boost::format fmt("%04d%02d%02d" "%02d%02d%02d");
  d_string += boost::str(boost::format(fmt) % (tm.tm_year+1900) % (tm.tm_mon+1) % tm.tm_mday % tm.tm_hour % tm.tm_min % tm.tm_sec);
```

File: pdns/ssqlite3.cc [hunk 5]
```diff
@@ -49,13 +49,14 @@ int pdns_sqlite3_clear_bindings(sqlite3_stmt *pStmt){
 class SSQLite3Statement: public SSqlStatement
 {
 public:
-  SSQLite3Statement(SSQLite3 *db, bool dolog, const string& query) : d_prepared(false)
+  SSQLite3Statement(SSQLite3 *db, bool dolog, const string& query) :
+    d_prepared(false),
+    d_query(query),
+    d_dolog(dolog),
+    d_stmt(NULL),
```
Comment (rgacogne) [inline pdns/ssqlite3.cc]: `nullptr`? :)

File: pdns/syncres.hh [hunk 6]
```diff
@@ -72,11 +72,8 @@ typedef map<
 template<class Thing> class Throttle : public boost::noncopyable
 {
 public:
-  Throttle()
+  Throttle() : d_limit(3), d_ttl(60), d_last_clean(time(0))
```
Comment (rgacogne) [inline pdns/syncres.hh]:
```suggestion
  Throttle() : d_limit(3), d_ttl(60), d_last_clean(time(nullptr))
```

File: pdns/histog.hh (hunk 1 above)
Comment (chbruyand) [reply pdns/histog.hh]: which should not work there :/

**Comment by rgacogne** [comment]
LGTM, although the branch has conflicts now :-/

**Comment by chbruyand** [comment]
Merged with last changes.

**Pull request merged**